### ISSUE TITLE
Recipe to search and  report of relocated dependencies

### ISSUE THREAD

**Issue by yeikel** [openrewrite/rewrite-migrate-java#291]
## What problem are you trying to solve?

In maven, there is a concept of relocation where library maintainers can publish a version of an artifact and provide additional metadata explaining the coordinate relocation : 

```xml

<project>
<modelVersion>4.0.0</modelVersion>
<groupId>ant</groupId>
<artifactId>ant</artifactId>
<version>1.7.0</version>
<distributionManagement>
<relocation>
<groupId>org.apache.ant</groupId>
</relocation>
</distributionManagement>
</project>
```
https://repo.maven.apache.org/maven2/ant/ant/1.7.0/ant-1.7.0.pom

In this case, ant moved from the group id `ant` ->  `org.apache.ant` so users would normally need to change this manually

![image](https://github.com/openrewrite/rewrite-migrate-java/assets/8935151/be13dbb0-3a5e-4041-bafa-e02f5a001af4)

## Describe the solution you'd like

I would like a search recipe to scan my direct dependencies in my pom file, detect and report the dependencies where this happened. We can create a separate recipe to also do the change but that's outside of the scope of this recipe as that's likely more complex. An example is described here #289 

## Have you considered any alternatives or workarounds?

I am not aware of any alternative. Users need to watch out for this and apply the upgrade manually. 

## Additional context

These changes tend to puts users at risks. it would be ideal to identity these changes automatically as part of the build process/a search recipe

Maven guide : https://maven.apache.org/guides/mini/guide-relocation.html

**Comment by timtebeek** [comment]
I like this suggestion, and would definitely help to get folks un-stuck from older dependencies. Thanks!

**Comment by yeikel** [comment]
> I like this suggestion, and would definitely help to get folks un-stuck from older dependencies. Thanks!

Definitely. Awareness is the hardest part, and  finding the newer coordinates is usually detective work

**Comment by timtebeek** [comment]
Had a quick look; we don't yet map `distributionManagement`, nor the `relocation` within that, which has [fields for `groupId`, `artifactId`, `version` and `message`](https://maven.apache.org/ref/3.9.4/maven-model/maven.html#relocation). Not sure if that's necessary, but I'd think so since the dependency poms are not Xml documents we parse as source files, but only into the model of `org.openrewrite.maven.internal.RawPom` and `org.openrewrite.maven.tree.Pom` that we pass around.
https://github.com/openrewrite/rewrite/blob/abf6339c0b26e0c40455c3de7bcd41d56471f339/rewrite-maven/src/main/java/org/openrewrite/maven/internal/RawPom.java#L50
https://github.com/openrewrite/rewrite/blob/abf6339c0b26e0c40455c3de7bcd41d56471f339/rewrite-maven/src/main/java/org/openrewrite/maven/tree/Pom.java#L41

I'll move this ticket to openrewrite/rewrite, since it looks to best fit in there given the search recipes we already have there.
